tests/adam_test.cpp

```diff
@@ -418,3 +418,44 @@ TEST_CASE("PadamLogisticRegressionTest", "[AdamTest]")
       coordinates);
   REQUIRE(testAcc == Approx(100.0).epsilon(0.006)); // 0.6% error tolerance.
 }
+
+TEST_CASE("SimpleAdamWTestFunction", "[AdamTest]")
+{
+  // Sometimes this test can fail randomly, so we allow it to run up to three
+  // times.
```

Comment:
This comment doesn't seem accurate.  We don't seem to be running this test three times.